lib/hexdocs/search/typesense.ex

```diff
@@ -0,0 +1,76 @@
+defmodule Hexdocs.Search.Typesense do
+  require Logger
+
+  @behaviour Hexdocs.Search
+
+  @collection "hexdocs"
+
+  @impl true
+  def index(package, version, search_items) do
+    full_package = "#{package}-#{version}"
+
+    ndjson =
+      Enum.map(search_items, fn item ->
+        json =
+          Map.take(item, ["type", "ref", "title", "doc"])
+          |> Map.put("package", full_package)
+          |> Jason.encode_to_iodata!()
+
+        [json, ?\n]
+      end)
+
+    url = url("collections/#{@collection}/documents/import?action=create")
+    headers = headers([{"content-type", "text/plain"}])
+
+    case :hackney.post(url, headers, ndjson) do
+      {:ok, 200, _resp_headers, ref} ->
```

Comment:
My understanding is we're later reading the entire response body so we might as well do it in one go:

```suggestion
    case :hackney.post(url, headers, ndjson, [:with_body]) do
      {:ok, 200, _resp_headers, body} ->
```

that being said, can we use `Hexdocs.HTTP` for consistency? If it's missing some features for this use case and it's not too much hassle we'd appreciate adjusting it but I wouldn't sweat it too much, I'd like to eventually use Req here anyway.

Reply:
We read the entire response but we don't persist it. I guess it's small enough to be read in full, I made the request "eager" in https://github.com/hexpm/hexdocs/pull/44/commits/0c6f014df0a714e8d505eb551a96c0a1d538ce16